### ISSUE TITLE
Using own data

### ISSUE THREAD

**Issue by theveloped** [AutodeskAILab/UV-Net#4]
thanks for sharing the amazing work done on UV-net. I'm very eager to do some tests on my own data and was wondering if you could give me some pointers on getting started. I'm familiar with pythonOCC and adjacency graphs, however fairly new to deep learning.

**Comment by pradeep-pyro** [comment]
Hi, Thanks for your interest. To train on custom data, the idea is to create face-adjacency graphs with UV-grids as node and edge features and save them in a format that is understood by UV-Net i.e., [DGL](https://www.dgl.ai/) based graphs in `.bin` format (there is a brief note in the readme [here](https://github.com/AutodeskAILab/UV-Net#data)).

Once the data is available in this format, a dataloader has to be written based on the existing examples in the `datasets` folder. For classification, the graph itself and a tensor containing the label index has to be returned (see [here](https://github.com/AutodeskAILab/UV-Net/blob/main/datasets/solidletters.py#L90)). For face segmentation, the per-face label index has to be stored in the graph as a node attribute named "y" (see [here](https://github.com/AutodeskAILab/UV-Net/blob/main/datasets/mfcad.py#L67)). Finally, the training/testing scripts have to modified to account for your new dataset. E.g., you may want to add a new item to the `--dataset` argument [here](https://github.com/AutodeskAILab/UV-Net/blob/main/classification.py#L14).

The data processing to convert solid models to `.bin` files for the paper was done using a proprietary solid modeling kernel and format. We have recently been moving all this pre-processing to pythonOCC so that an end-user can run a Python script on a folder of `.step` files and get the `.bin` files in a format that can be fed to UV-Net.

This needs a bit more testing before release, and will be done soon.
